lib/misc.py

```diff
@@ -66,6 +66,10 @@ def delete(cli):
    if (delete_ct == 0):
       ch.FATAL("no image matching glob, can’t delete: %s" % cli.image_ref)
    bu.cache.worktrees_fix()
+   if (not cli.not_cache):
+      to_delete = cli.image_ref.replace("/", "%").replace(":", "+") + "#"
+      bu.cache.branch_delete(to_delete)
+      bu.cache.branch_delete(to_delete[:-1])
```

Comment:
1. Probably clearer to build up the branch names constructively, rather than destructively.
2. There are existing methods to convert an image ref to a branch name, whether or not ready, I'm pretty sure.